OpenRA.Mods.Common/Traits/Player/StorageWarning.cs

```diff
@@ -0,0 +1,60 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Provides the player with an audible warning when their storage is nearing full.")]
+	public class StorageWarningInfo : ITraitInfo, Requires<PlayerResourcesInfo>
+	{
+		[Desc("Interval at which to check if more storage is needed.")]
+		public readonly int AdviceInterval = 250;
+
+		[Desc("The threshold (as a ratio of resources to capacity) above which a warning is played.")]
+		public readonly float Threshold = 0.8f;
+
+		[Desc("The sound type of the warning.")]
+		public readonly string SoundType = "Speech";
+
+		[Desc("Sound to play for the warning.")]
+		public readonly string SoundName = "SilosNeeded";
```

Comment:
Please make this `Notification` and then remove the `SoundType` field above.  See the comments in #9605 for an explanation as to why a toggle between speech and notifications is bad.